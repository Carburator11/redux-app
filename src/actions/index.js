import * as actionType from './ActionType';

export const addCounter = () => ({
  type: actionType.ADD_COUNTER,
  payload: 1
});

export const removeCounter = () => ({
  type: actionType.REMOVE_COUNTER,
  payload: 1
});

export const addTodo = (text) => ({
  type: actionType.ADD_TODO,
  text
});

export const removeTodo = (index) => ({
  type: actionType.REMOVE_TODO,
  index
});

export const toggleComplete = (index) => ({
  type: actionType.TOGGLE_COMPLETE,
  index
});
