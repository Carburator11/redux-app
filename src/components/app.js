import React from 'react'

// Counter
import Counter from '../container/counter/'
import Incr    from '../container/counter/increment'
import Decr    from '../container/counter/decrement'
import Reset   from '../container/counter/reset'

// Todo app
import AddTodo from '../container/todo/addTodo'
import FilterList  from '../container/todo/filterList'
import FilterLink from '../container/todo/filterLink'

/* This is just a presentationnal Component */
const App = () => {
    return(
        <div style = {{'margin': 40}}>
            <h1>My redux app</h1>
            <h2>Counter</h2>
            <Counter/><br />
            <Incr />
            <Decr />
            <Reset />
            <h2>To do List</h2>
            <AddTodo />
            <FilterList />
            <FilterLink />
        </div>
    )}

export default App