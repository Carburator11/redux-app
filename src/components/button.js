import React from 'react';

const Button = ({placeholder, onClick}) => <button onClick = {onClick}>{placeholder}</button>

export default Button;