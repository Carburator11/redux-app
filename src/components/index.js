import App from './app';
import Button from './button';
import ReduxCounter from './reduxCounter'
import LocalCounter from './localCounter';

export { App, Button, ReduxCounter, LocalCounter }