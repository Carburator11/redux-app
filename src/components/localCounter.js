import React from 'react';
import Button from './button';

const LocalCounter = ({count, incr, decr}) => (
    <div>
        <h4>Local counter : {count}</h4>
        <Button placeholder="-" onClick={decr}/>
        <Button placeholder="+" onClick={incr}/>
    </div>)

export default LocalCounter