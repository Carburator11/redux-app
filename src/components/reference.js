import React from 'react';

const Reference = () =>
    <div style={{'marginTop': 20}}>
        <h4>Documentation</h4>
        &#8226; <a href="https://facebook.github.io/jest/en/">Jest</a><br/>
        &#8226; <a href="http://airbnb.io/enzyme/docs/api/">Enzyme</a><br/>
        
        <h4>Biblio</h4>
        &#8226; <a href="https://youtu.be/Eu35xM76kKY">Unit Testing in JS [video]</a><br/>
        &#8226; <a href="https://hackernoon.com/testing-react-components-with-jest-and-enzyme-41d592c174f">Testing React components with Jest and Enzyme</a><br/>
        &#8226; <a href="https://hackernoon.com/problematic-react-lifecycle-methods-are-going-away-in-react-17-4216acc7d58b">Problematic React Lifecycle Methods are Going Away in React 17</a><br/>
    </div>


export default Reference