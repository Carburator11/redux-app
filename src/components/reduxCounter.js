import React from 'react';
import { connect } from 'react-redux';
import { decrementCounter, incrementCounter } from '../actions/';
import Button from './button';

const Counter = ({ reduxCount, decrementCounter, incrementCounter}) => (
    <div>
        <h4> Redux counter :  {reduxCount}</h4> 
            <Button 
                key="decr" 
                placeholder="-" 
                onClick={ e => {
                    e.preventDefault();
                    decrementCounter() 
            }} />
            <Button
                key="incr"
                placeholder="+"
                onClick={ e => {
                        e.preventDefault();
                        incrementCounter()
            }} />
    </div>)

const mapStateToProps = ({ reduxCount }) => ({ reduxCount })

const mapDispatchToProps = dispatch => ({
    decrementCounter: () => dispatch( decrementCounter()),
    incrementCounter: () => dispatch( incrementCounter())     
})

export default connect(mapStateToProps, mapDispatchToProps)(Counter);