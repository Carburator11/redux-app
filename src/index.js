
import React from 'react';
import ReactDOM from 'react-dom';
//import { Router, Route, hashHistory } from 'react-router';
import { createStore } from 'redux';
import { Provider } from 'react-redux'
import { addCounter } from './actions/'
import reducers from './reducer/'
import App from './components/app'

let store = createStore( reducers );

ReactDOM.render(
	<Provider store={store}>
        <App />
	</Provider>,
	document.getElementById('app')
);
// Router :  {/*<Router history={hashHistory}>{routes}</Router>*/}


// Test..
// ReactDOM.render(<h1>Hello worldzzz !!!</h1>, document.getElementById('app'))
