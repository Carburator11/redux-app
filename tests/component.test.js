import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ReduxCounter, LocalCounter, Button } from '../src/components/';
 import configureStore from 'redux-mock-store';

test('LocalCounter should render two <Button /> components', () => {
    const wrapper = shallow(<LocalCounter />)
    expect(wrapper.find(Button)).toHaveLength(2)
});

test('ReduxCounter should render two <Button /> components', () => {
    const mockStore = configureStore({reduxCount: 0}); 
    const store = mockStore({})
    const wrapper = mount(<ReduxCounter store={store} />)
    expect(wrapper.find(Button)).toHaveLength(2)
});

test('LocalCounter snapshot', ()=>{
    const tree = renderer
    .create(<LocalCounter />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

test('ReduxCounter snapshot', ()=>{
    const mockStore = configureStore({reduxCount: 0});     
    const store = mockStore({})
    const tree = renderer
    .create(<ReduxCounter store={store}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
})