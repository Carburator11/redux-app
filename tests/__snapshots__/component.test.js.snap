// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalCounter snapshot 1`] = `
<div>
  <h4>
    Local counter : 
  </h4>
  <button
    onClick={undefined}
  >
    -
  </button>
  <button
    onClick={undefined}
  >
    +
  </button>
</div>
`;

exports[`ReduxCounter snapshot 1`] = `
<div>
  <h4>
     Redux counter :  
  </h4>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
